datalad/utils.py

```diff
@@ -2590,3 +2591,41 @@ def check_symlink_capability(path, target):
             path.unlink()
         if target.exists():
             target.unlink()
+
+
+_cp_supports_reflink = None
+
+
+def copy_file(src, dst, follow_symlinks=True):
+    """
+    Copy file (including metadata) from src to dst.  Use `cp --reflink=auto` if
+    the local `cp` supports it.
+
+    Parameters
+    ----------
+    src: str or os.PathLike
+      path to the file to copy
+    dst: str or os.PathLike
+      path that `src` should be copied to
+    """
+    global _cp_supports_reflink
+    if _cp_supports_reflink is None:
```

Comment:
Shouldn't this be made conditional to POSIX systems? Windows is already a slow target, and there is no chance that it could come back with a positive result.

Reply:
wouldn't this then become yet another OS-specific, vs scenario (tool, filesystem, etc) specific decisions we are trying to avoid?

Reply:
e.g. some https://en.wikipedia.org/wiki/ReFS from microsoft seems to aim to support CoW...

Reply:
I doubt that MS will expose CoW through win32 version of a POSIX command. With this change, the first call for each process will either determine that `cp` is not a command (on any standard windows system), or will use a command that is not from the OS. In both cases there is likely not benefit, and the cost is unconditional -- on windows.

Reply:
DataLad uses `git` and `git-annex`. `cp` is shipped with `git` on Windows, and AFAIK is used at least by `git-annex`, so I expect POSIX `cp` to be available to every DataLad installation including Windows.  As for doubts and hopes of CoW support -- we would not know until we try, and can only see if it is not supported today, but situation might change tomorrow. so I still do not see the reason for limiting besides making sure that this function works correctly with Windows style paths.

Reply:
`cp` is not in the PATH of cmd.exe when Git is installed with defaults via the installer and git-annex too. Both work nevertheless.

I am not sure, why `not on_windows` is problematic in this very spot.